:sparkles: adding solution server scenario - expanded version (#832)
konveyor/kai: docs/scenarios/migrating_custom_library_apps_using_solution_server/Migrating_inventory_management.md

# Modernizing Healthcare Applications: Audit Library Migration Using Konveyor AI and Solution Server

## Goal

> This scenario demonstrates how **Konveyor AI (Kai)** can assist in modernizing **two healthcare applications** by migrating from an outdated audit logging library (v1) to a modern streaming-based audit library (v2). We will use custom rules with Kai to analyze both applications and generate refactoring recommendations, then leverage the **Solution Server** to demonstrate how migration patterns learned from the first application can be automatically applied to the second application.

Reviewer: IMHO this is the main scenario. suggestion dont lead with migration of two healthcare applications. 
```suggestion
Migrate an outdated audit logging library (v1) to a modern streaming-based audit library (v2). We will use custom rules with Kai to analyze both applications and generate refactoring recommendations. Furthermore we will use the **Solution Server**, through which we demonstrate how migration patterns learned from the first application can be automatically applied to the second application.
```

## Applications to Modernize

> 

Reviewer: Add a mermaid drawing
```suggestion
graph TD
    subgraph "Core Systems"
        Inventory["Medical Device Inventory Management System"]
        EHR["Electronic Health Record (EHR) System"]
    end

    subgraph "Shared Dependencies"
        AuditLib("v1 Audit Library")
        SpringBoot("Spring Boot 2 (Java 8)")
    end

    subgraph "System-Specific Features"
        InventoryFeatures("REST API Endpoints<br/>In-memory Storage")
        EHRFeatures("Manages Patient Data<br/>Logs User Actions (login, logout, etc.)")
    end

    Inventory -- "Built with" --> SpringBoot
    EHR -- "Built with" --> SpringBoot

    Inventory -- "Uses for Logging" --> AuditLib
    EHR -- "Uses for Logging" --> AuditLib

    Inventory -- "Has Features" --> InventoryFeatures
    EHR -- "Has Features" --> EHRFeatures
```

1. **[Medical Device Inventory Management System](https://github.com/savitharaghunathan/inventory_management)** - Manages medical device inventory with comprehensive audit logging. Built with Spring Boot 2 (Java 8) and features REST API endpoints for medical device operations, in-memory storage with sample inventory data, and audit logging using the v1 audit library.

2. **[Electronic Health Record (EHR) System](https://github.com/hhpatel14/ehr_viewer/tree/java8)** - Patient data management system that also uses the same audit logging library for compliance tracking. Built with Spring Boot 2 (Java 8), featuring audit logging for all user actions including login, logout, and patient record access.

> This approach showcases the power of Kai's Solution Server in learning from one migration and applying those patterns to similar applications, significantly reducing migration effort and ensuring consistency across healthcare systems. The Solution Server delivers two primary benefits: **Contextual Hints** that surface examples of past migration solutions—including successful user modifications and accepted fixes—enabling Kai to offer actionable hints for difficult or previously unsolved migration problems; and **Migration Success Metrics** that expose detailed success metrics for each migration rule, derived from real-world usage data, enabling IDEs and automation tools to present users with confidence levels for successful migration of given code segments.

Reviewer: ```suggestion
## What is a Solution Server?

Kai's Solution Server systematically reduces migration effort and ensures code consistency by applying learned transformation patterns across similar applications. The server provides two primary functions derived from historical migration data:

1.  **Contextual Hints:** For complex or previously unsolved migration issues, the server provides actionable solutions by surfacing proven examples from past migrations, including user-modified and accepted code transformations.

2.  **Migration Success Metrics:** The server calculates and exposes success metrics for each migration rule based on aggregated usage data. This enables IDEs and automation tools to present a quantitative confidence level for the successful transformation of any given code segment.
```

## Applications to Modernize
```mermaid
graph TD
    subgraph "Core Systems"
        Inventory["Medical Device Inventory Management System"]
        EHR["Electronic Health Record (EHR) System"]
    end

    subgraph "Shared Dependencies"
        AuditLib("v1 Audit Library")
        SpringBoot("Spring Boot 2 (Java 8)")
    end

    subgraph "System-Specific Features"
        InventoryFeatures("REST API Endpoints<br/>In-memory Storage")
        EHRFeatures("Manages Patient Data<br/>Logs User Actions (login, logout, etc.)")
    end

    Inventory -- "Built with" --> SpringBoot
    EHR -- "Built with" --> SpringBoot

    Inventory -- "Uses for Logging" --> AuditLib
    EHR -- "Uses for Logging" --> AuditLib

    Inventory -- "Has Features" --> InventoryFeatures
    EHR -- "Has Features" --> EHRFeatures
1. **[Medical Device Inventory Management System](https://github.com/savitharaghunathan/inventory_management)** - Manages medical device inventory with comprehensive audit logging. Built with Spring Boot 2 (Java 8) and features REST API endpoints for medical device operations, in-memory storage with sample inventory data, and audit logging using the v1 audit library.

> 

Author: done
